src/lib.rs

```diff
@@ -243,31 +243,41 @@ impl<T> Sending<T> {
 
 	fn push(&mut self, item: T) {
 		self.items.push_back(item);
-		self.flushing = false;
```

Comment:
Note that I removed `flushing` since I don't see its purpose and it may interfere with the sending logic.